src/host/handle_ring_buffer.h

```diff
@@ -48,13 +48,38 @@ namespace asynchost
              thread_id,
              msg] = ringbuffer::read_message<AdminMessage::log_msg>(data, size);
 
-          logger::Out::write(
-            file_name,
-            line_number,
-            log_level,
-            thread_id,
-            msg,
-            log_time_us_count);
+          logger::LogLine ll(
+            log_level, file_name.c_str(), line_number, thread_id);
+          ll.msg = msg;
+
+          // Represent offset as a real (counting seconds) to handle both small
+          // negative _and_ positive numbers. Since the system clock used is not
+          // monotonic, the offset we calculate could go in either direction,
```

Comment:
What's the reason we don't we use `steady_clock` again?

Reply:
I'm not sure. Perhaps something to do with converting to printable C types? "It is the only C++ clock that has the ability to map its time points to C-style time." But I don't see a strong requirement for it.